distributed/comm/addressing.py

```diff
@@ -119,15 +119,24 @@ def get_address_host(addr):
 
     In contrast to get_address_host_port(), this function should always
     succeed for well-formed addresses.
-
-    >>> get_address_host('tcp://192.168.10.50:8687')
-    '192.168.10.50'
     """
     scheme, loc = parse_address(addr)
     backend = registry.get_backend(scheme)
     return backend.get_address_host(loc)
 
 
+def get_local_address_for(addr):
+    """
+    Get the local listening address suitable for reaching *addr*.
+
+    For instance, trying to reach an external TCP address will return
+    a local TCP address that's routable to that external address.
+    """
```

Comment:
I'm not sure I have the expertise in networking to provide an accurate definition.  The examples you've provided would probably help.  

Generally this might be a nice pattern to repeat throughout `distributed/comm/addressing.py`.  When I look at this file I find myself daunted by the variety of functions here that do slightly different things.